src/CookieConsent.tsx

```diff
@@ -113,4 +113,15 @@ const CookieConsentProvider: React.FC<React.PropsWithChildren<Props>> = ({
   );
 };
 
-export { CookieConsentProvider, CookieConsentContext };
+const useCookieConsent = () => {
+  const context = React.useContext(CookieConsentContext);
+  if (context === undefined) {
+    throw new Error(
+      'useCookieConsent must be used within a CookieConsentProvider'
+    );
+  }
+
+  return context;
+};
+
+export { CookieConsentProvider, CookieConsentContext, useCookieConsent };
```

Comment:
I'd go for not exporting the context here and making it a breaking change

Reply:
In that case, we would disable React option to access context via `Context.Consumer`.